mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoscalerManager.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.worker.jobmaster;
+
+import java.util.Properties;
+
+/**
+ * Failover status client interface to get the failover status of current region
+ */
+public interface JobAutoscalerManager {
```

Comment:
what's the relationship between JobAutoscalerManager and the existing auto-scaling strategies? Does it apply on top of all the strategies?

Reply:
yep, there's a way to enable the autoscaler manager for a job or jobcluster. If enabled, yes it'll apply on top of all autoscaling strategies.